File: etamen-backend/app/Modules/Medications/Infrastructure/Models/MedicationReminderTime.php
<?php

namespace App\Modules\Medications\Infrastructure\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class MedicationReminderTime extends Model
{
    protected $fillable = [
        'medication_reminder_id',
        'time_of_day',
        'label',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
        ];
    }

    public function reminder(): BelongsTo
    {
        return $this->belongsTo(MedicationReminder::class, 'medication_reminder_id');
    }
}

File: etamen-flutter/lib/features/medications/domain/entities/medication_reminder_time.dart
class MedicationReminderTime {
  const MedicationReminderTime({
    required this.id,
    required this.medicationReminderId,
    required this.timeOfDay,
    this.label,
    this.isActive = true,
    this.createdAt,
    this.updatedAt,
  });

  final int id;
  final int medicationReminderId;
  final String timeOfDay;
  final String? label;
  final bool isActive;
  final DateTime? createdAt;
  final DateTime? updatedAt;
}
